test/models/financialDetails/ChargeItemSpec.scala

```diff
@@ -65,6 +70,165 @@ class ChargeItemSpec extends UnitSpec with ChargeConstants  {
     override def getCurrentDate: LocalDate = currentDate
   }
 
+  "ChargeItem" when {
+
+    "isOverdueReviewAndReconcileAccruingInterestCharge method" when {
+
+      "transaction type is PoaOneReconciliationDebit, is not overdue and is not paid returns true" in {
+        val dateServiceBeforeDueDate = dateService(dueDate.minusDays(1))
+
+        val chargeItem = ChargeItem.fromDocumentPair(
+          documentDetail = defaultDocDetails,
+          financialDetails = List(PoaOneReconciliationDebitDetails))
+
+
+        chargeItem.isOverdueReviewAndReconcileAccruingInterestCharge()(dateServiceBeforeDueDate) shouldBe true
+      }
+
+      "transaction type is PoaTwoReconciliationDebit, is not overdue and is not paid returns true" in {
+        val dateServiceBeforeDueDate = dateService(dueDate.minusDays(1))
+
+        val chargeItem = ChargeItem.fromDocumentPair(
+          documentDetail = defaultDocDetails,
+          financialDetails = List(PoaTwoReconciliationDebitDetails))
+
+        chargeItem.isOverdueReviewAndReconcileAccruingInterestCharge()(dateServiceBeforeDueDate) shouldBe true
+      }
+
+      "returns false when charge item is not PoaOneReconciliationDebit or PoaTwoReconciliationDebit " in {
+        val dateServiceAfterDueDate = dateService(dueDate.minusDays(1))
+
+        val chargeItem = ChargeItem.fromDocumentPair(
+          documentDetail = defaultDocDetails,
+          financialDetails = List(balancingNics2FinancialDetails))
+
+        chargeItem.isOverdueReviewAndReconcileAccruingInterestCharge()(dateServiceAfterDueDate) shouldBe false
+      }
+
+      "charge is overdue and is not paid returns false" in {
+        val dateServiceAfterDueDate = dateService(dueDate.plusDays(1))
+
+        val chargeItem = ChargeItem.fromDocumentPair(
+          documentDetail = defaultDocDetails,
+          financialDetails = List(PoaOneReconciliationDebitDetails))
+
+        chargeItem.isOverdueReviewAndReconcileAccruingInterestCharge()(dateServiceAfterDueDate) shouldBe false
+      }
+
+      "charge is not overdue and is paid returns false" in {
+        val dateServiceBeforeDueDate = dateService(dueDate.minusDays(1))
+
+        val chargeItem = ChargeItem.fromDocumentPair(
+          documentDetail = docDetailsNoOutstandingAmout,
+          financialDetails = List(PoaOneReconciliationDebitDetails))
+
+        chargeItem.isOverdueReviewAndReconcileAccruingInterestCharge()(dateServiceBeforeDueDate) shouldBe false
+      }
+
+    }
+
+    "getDueDate method" when {
```

Comment:
usually no need to mention "_method_" in the test scenario / as closer you specifying business need as better ;